### ISSUE TITLE
False positive `used-before-assignment` with try/else/continue

### ISSUE THREAD

**Issue by mattclay** [pylint-dev/pylint#6804]
### Bug description

The `used-before-assignment` check reports a false positive with try/else/continue starting with pylint 2.13.0.

This issue is nearly identical to https://github.com/PyCQA/pylint/issues/6790, except it involves `continue` instead of `return`.

```python
"""bug.py"""


def demo():
    """Demonstrate used-before-assignment regression in pylint==2.13.0 and 2.14.0."""
    for _temp in range(0, 1):
        try:
            pass
        except ValueError:
            error = True
        else:
            continue

        print(error)
```

### Command used

```shell
pylint bug.py
```

### Pylint output

```shell
************* Module bug
bug.py:14:14: E0601: Using variable 'error' before assignment (used-before-assignment)
```

### Expected behavior

No errors reported, as was the case for 2.12.2.

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.9.5 (default, Nov 23 2021, 15:27:38) 
[GCC 9.3.0]
```

### OS / Environment

Ubuntu 22.04.4

**Comment by jacobtylerwalls** [comment]
This one's harder because depending on `error` is not safe if attempted outside the loop, so my hope would be to distinguish the two cases.
